packages/aws-rfdk/lib/deadline/lib/repository.ts

```diff
@@ -581,20 +581,22 @@ export class Repository extends Construct implements IRepository {
     if (props.fileSystem instanceof MountableEfs && !props.fileSystem.accessPoint) {
       throw new Error('When using EFS with the Repository, you must provide an EFS Access Point');
     }
+    if ((props.secretsManagementSettings?.enabled ?? true) && props.database && !props.database.databaseConstruct) {
+      throw new Error('Admin credentials for Deadline Secrets Management cannot be generated when using imported database. Please provide secret with admin credentials or disable Secrets Management in repository properties.');
+    }
 
     this.version = props.version;
 
     this.secretsManagementSettings = {
       enabled: props.secretsManagementSettings?.enabled ?? true,
       credentials: props.secretsManagementSettings?.credentials ??
-        ((props.secretsManagementSettings?.enabled ?? true) ? new Secret(this, 'SMAdminUser', {
+        ((props.secretsManagementSettings?.enabled ?? true) ? new Secret( props.database?.databaseConstruct ?? this, 'SMAdminUser', {
```

Comment:
I'm not sure whether we want to put the secret in the database's scope or in the stack's scope. The alternative is:

```suggestion
        ((props.secretsManagementSettings?.enabled ?? true) ? new Secret(Stack.of(props.database?.databaseConstruct) ?? this, 'SMAdminUser', {
```

We need to commit to whatever choice  we make here, since changing scopes is a breaking change. The choice of scopes has two potential impacts that I'm aware of:

## Identifiers

In your current choice of scope, the secret's construct path derives from the construct path of the database. The logical IID for the secret becomes a compound value of the construct path ([ref](https://docs.aws.amazon.com/cdk/latest/guide/identifiers.html)). Depending on the implementation of the `AWS::SecretsManager::Secret` resource provider, it may use the logical ID to generate the secret name. This will affect how readable the secret name is in the AWS console and API responses. That being said, if there 

## Aspects

[CDK's aspects](https://docs.aws.amazon.com/cdk/latest/guide/aspects.html) can be selectively applied to a node in the construct tree and its descendents. Users intending to walk the construct nodes that are children of the database will also visit the secret's construct nodes.  This may depend on the call order, but it is worth some consideration.

Reply:
I think putting it in the Stack's scope is probably the right place for it.

Reply:
> I think putting it in the Stack's scope is probably the right place for it.

I agree, although I think there is a logic error in my original suggestion. I think it should be:

```suggestion
        ((props.secretsManagementSettings?.enabled ?? true) ? new Secret(props.database?.databaseConstruct ? Stack.of(props.database.databaseConstruct) : this, 'SMAdminUser', {
```

Reply:
Looks good - thanks!